tensorflow_datasets/image/imagewang.py

```diff
@@ -0,0 +1,140 @@
+# coding=utf-8
+# Copyright 2020 The TensorFlow Datasets Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+""" Imagewang contains Imagenette and Imagewoof combined. """
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+
+import tensorflow as tf
+import tensorflow_datasets.public_api as tfds
+
+# TODO(imagewang): BibTeX citation
+# See https://github.com/fastai/imagenette/issues/16.
+_CITATION = """
+"""
+
+_DESCRIPTION = """\
+Imagewang contains Imagenette and Imagewoof combined
+Image网 (pronounced "Imagewang"; 网 means "net" in Chinese) contains Imagenette
+and Imagewoof combined, but with some twists that make it into a tricky
+semi-supervised unbalanced classification problem:
+
+* The validation set is the same as Imagewoof (i.e. 30% of Imagewoof images);
+  there are no Imagenette images in the validation set (they're all in the
+  training set)
+* Only 10% of Imagewoof images are in the training set!
+* The remaining are in the unsup ("unsupervised") directory, and you can not
+  use their labels in training!
+* It's even hard to type and hard to say!
+
+The dataset comes in three variants:
+  * Full size
+  * 320 px
+  * 160 px
+This dataset consists of the Imagenette dataset {size} variant.
+"""
+
+_LABELS_FNAME = "image/imagewang_labels.txt"
+_URL_PREFIX = "https://s3.amazonaws.com/fast-ai-imageclas"
+_SIZES = ["full-size", "320px", "160px"]
+
+_SIZE_TO_DIRNAME = {
+    "full-size": "imagewang",
+    "320px": "imagewang-320",
+    "160px": "imagewang-160"
+}
+
+
+class ImagewangConfig(tfds.core.BuilderConfig):
+  """BuilderConfig for Imagewang."""
+
+  def __init__(self, size, **kwargs):
+    super(ImagewangConfig, self).__init__(
+        version=tfds.core.Version("2.0.0"), **kwargs)
+    self.size = size
+
+
+def _make_builder_configs():
+  configs = []
+  for size in _SIZES:
+    configs.append(
+        ImagewangConfig(
+            name=size,
+            size=size,
+            description=_DESCRIPTION.format(size=size)))
```

Comment:
We have descriptions in two places:

the one in DatasetInfo (line 94): should be long and about whole dataset.
The one for the config - should be shorter and basically only chat about what this configuration is about (rather than about the whole dataset again).